tox.ini

```diff
@@ -29,6 +31,7 @@ deps =
 commands = pytest --pylint -m pylint {posargs}
 
 [testenv:typecheck]
+basepython = python3.7
```

Comment:
I would've thought the newer syntax in this PR would break on older Python versions, and this project wouldn't be able to adopt the syntax yet. However, the `from __future__ import annotations` at the top of source files that use the syntax seems to allow the syntax, as far back as Python 3.7.